### PR TITLE
[risk=low][RW-13311] SAS autodelete must be enabled

### PR DESCRIPTION
**Pull request by jmthibault79**
SAS Autodelete

---
**PR checklist**

- [x] I have included an issue ID or "no ticket" in the PR title as outlined in [CONTRIBUTING.md](https://github.com/all-of-us/workbench/blob/main/.github/CONTRIBUTING.md).
- [x] I have included a risk tag of the form `[risk=no|low|moderate|severe]` in the PR title as outlined in [CONTRIBUTING.md](https://github.com/all-of-us/workbench/blob/main/.github/CONTRIBUTING.md).
- [ ] I have manually tested this change and my testing process is described above.
- [x] This change includes appropriate automated tests, and I have documented any behavior that cannot be tested with code.
- [ ] I have added explanatory comments where the logic is not obvious.
- One or more of the following is true:
  - [ ] This change is intended to complete a JIRA story, so I have checked that all AC are met for that story.
  - [ ] This change fixes a bug, so I have ensured the steps to reproduce are in the Jira ticket or provided above.
  - [ ] This change impacts deployment safety (e.g. removing/altering APIs which are in use), so I have documented the impacts in the description.
  - [ ] This change includes a new feature flag, so I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later.
  - [x] This change modifies the UI, so I have taken screenshots or recordings of the new behavior and notified the PO and UX designer in [Slack](https://pmi-engteam.slack.com/archives/C02MWP2RN5P).
  - [ ] This change modifies API behavior, so I have run the relevant E2E tests locally because API changes are not covered by our PR checks.
  - [ ] None of the above apply to this change.

### CODE REVIEW

File: ui/src/app/components/gke-app-configuration-panels/create-gke-app.spec.tsx [hunk 1]
```diff
@@ -248,26 +253,6 @@ describe(CreateGkeApp.name, () => {
         expect(deleteButton).toBeNull();
       });
 
-      it('should allow disabling autodelete', async () => {
```
Comment (jmthibault79) [inline ui/src/app/components/gke-app-configuration-panels/create-gke-app.spec.tsx:251]: split into SAS and non-SAS versons

File: ui/src/app/components/gke-app-configuration-panels/create-gke-app.spec.tsx [hunk 2]
```diff
@@ -624,11 +609,10 @@ describe(CreateGkeApp.name, () => {
       });
 
       it(`Should allow to create app request for valid disk size for ${appType}`, async () => {
+        const spyCreateApp = getCreateAppSpy();
+
         // Arrange
         await component(appType);
-        const spyCreateApp = jest
```
Comment (jmthibault79) [inline ui/src/app/components/gke-app-configuration-panels/create-gke-app.spec.tsx:629]: can reuse other code

File: ui/src/app/components/gke-app-configuration-panels/create-gke-app.tsx [hunk 3]
```diff
@@ -355,18 +355,25 @@ export const CreateGkeApp = ({
...
-            aria-label='Auto-deletion toggle'
-            disabled={isAppActive(app)}
-            checked={createAppRequest.autodeleteEnabled}
-            onChange={(autodeleteEnabled: boolean) => {
-              setCreateAppRequest((prevState) => ({
-                ...prevState,
-                autodeleteEnabled,
-              }));
-            }}
-            style={{ marginRight: '1rem', zoom: 1.5 }}
```
Comment (jmthibault79) [inline ui/src/app/components/gke-app-configuration-panels/create-gke-app.tsx:368]: moving this centers the tooltip

**Review by evrii — APPROVED** [review]
Tested locally and worked as expected.The only thing that might be worth addressing is the info icon tooltip:
Screenshot 2024-09-10 at 9 52 33 AM
The section about disabling autodelete doesn't fit for SAS anymore.

**Comment by jmthibault79** [comment]
Good idea.  I removed that section for SAS only.

Screenshot 2024-09-10 at 11 44 36 AM

**Pull request merged**